File: solutions/scala/leap/1/src/main/scala/Leap.scala

def is_divisible_by(n: Int, target: Int): Boolean = n % target == 0

object Leap {
  def leapYear(year: Int): Boolean = is_divisible_by(year, 4) && (!is_divisible_by(year, 100) || is_divisible_by(year, 400))
}

File: solutions/kotlin/leap/1/src/main/kotlin/Leap.kt
data class Year(private val year: Int) {
    val isLeap: Boolean =
        year.isDivisibleBy(4) && (!year.isDivisibleBy(100) || year.isDivisibleBy(400))

    private fun Int.isDivisibleBy(divisor: Int): Boolean = this % divisor == 0
}
